lib/streamlit/bootstrap.py

```diff
@@ -62,15 +62,29 @@ def _fix_sys_path(script_path):
 def _fix_matplotlib_crash():
     """Set Matplotlib backend to avoid a crash.
 
-    The default Matplotlib backend crashes Python for most MacOS users.
-    So here we set a safer backend as a fix. Users can always disable this
-    behavior by setting the config runner.fixMatplotlib = false.
+    The default Matplotlib backend crashes Python on OSX when run on a thread
+    that's not the main thread, so here we set a safer backend as a fix.
+    Users can always disable this behavior by setting the config
+    runner.fixMatplotlib = false.
 
     This fix is OS-independent. We didn't see a good reason to make this
     Mac-only. Consistency within Streamlit seemed more important.
     """
     if config.get_option("runner.fixMatplotlib"):
```

Comment:
Add a check for whether this is a Mac.

Reply:
Ok, I just submitted a fix for this. Plus tests.

Reply:
Turns out this is needed on Linux in CircleCI Py2 in our tests, otherwise Matplotlib crashes saying there's no $DISPLAY set. Since I don't want to investigate it right now, I'm leaving this as is.